### PR TITLE
Voting Manager fix to ignore duplicate collateral members (1.1.0.0)

### PR DESCRIPTION
**Pull request by quantumagi**
PR #272 specified that voting, which would otherwise result in adding a member, would not do so if collateral address re-use is detected.

Similarly (this PR) such members should also be omitted when the federation is later re-created from the polls.

On a side-note, a follow-up PR is required to somehow strengthen/fix the `VotingRequestValidationRule` to prevent such duplicate addresses from appearing in voting requests when already in a voting request, the active federation, pending or approved polls. I.e. these duplications should never reach the VM in the first place.

### CODE REVIEW

**Comment by quantumagi** [comment]
image

File: src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs [hunk 1]
```diff
@@ -276,9 +276,14 @@ public List<IFederationMember> GetFederationFromExecutedPolls()
                     IFederationMember federationMember = ((PoAConsensusFactory)(this.network.Consensus.ConsensusFactory)).DeserializeFederationMember(poll.VotingData.Data);
 
                     if (poll.VotingData.Key == VoteKey.AddFederationMember)
-                        federation.Add(federationMember);
+                    {
+                        if (!federation.Any(m => m is CollateralFederationMember colMember && federationMember is CollateralFederationMember colMember2 && colMember.CollateralMainchainAddress == colMember2.CollateralMainchainAddress))
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs:280]: Shouldnt the `federationMember is CollateralFederationMember colMember2` part here be outside of the `Any` ?

It probs amounts to the same but it reads better and also the type doesnt happen for every member in the federation?

Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs:280]: Or is this acting as a compare then?

Comment (quantumagi) [reply src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs]: Good spot

**Review by fassadlr — APPROVED** [review]

**Comment by quantumagi** [comment]
We need a PR to bump the NBitcoin version. Some other PR somehow got merged with a change to NBitcoin without bumping the version...

**Pull request merged**